cg/cli/workflow/commands.py

```diff
@@ -0,0 +1,188 @@
+import logging
+import shutil
+from pathlib import Path
+import datetime as dt
+
+import click
+from dateutil.parser import parse as parse_date
+
+from cg.constants import EXIT_SUCCESS, EXIT_FAIL
+from cg.exc import FlowcellsNeededError
+from cg.meta.workflow.analysis import AnalysisAPI
+from cg.meta.workflow.balsamic import BalsamicAnalysisAPI
+from cg.meta.workflow.mip_dna import MipDNAAnalysisAPI
+
+OPTION_DRY = click.option(
+    "-d", "--dry-run", "dry_run", help="Print command to console without executing", is_flag=True
+)
+ARGUMENT_CASE_ID = click.argument("case_id", required=True)
+
+LOG = logging.getLogger(__name__)
+
+
+@click.command("ensure-flowcells-ondisk")
+@ARGUMENT_CASE_ID
+@click.pass_context
+def ensure_flowcells_ondisk(context: click.Context, case_id: str):
+    """Check if flowcells are on disk for given case. If not, request flowcells and raise FlowcellsNeededError"""
+    analysis_api: AnalysisAPI = context.obj["analysis_api"]
+    analysis_api.verify_case_id_in_statusdb(case_id=case_id)
+    if not analysis_api.all_flowcells_on_disk(case_id=case_id):
+        raise FlowcellsNeededError(
+            "Analysis cannot be started: all flowcells need to be on disk to run the analysis"
+        )
+    LOG.info("All flowcells present on disk")
+
+
+@click.command("resolve-compression")
+@ARGUMENT_CASE_ID
+@OPTION_DRY
+@click.pass_context
+def resolve_compression(context: click.Context, case_id: str, dry_run: bool):
+    """Handles cases where decompression is needed before starting analysis"""
+    analysis_api: AnalysisAPI = context.obj["analysis_api"]
+    analysis_api.verify_case_id_in_statusdb(case_id=case_id)
+    analysis_api.resolve_decompression(case_id=case_id, dry_run=dry_run)
+
+
+@click.command()
+@ARGUMENT_CASE_ID
+@click.pass_context
+def link(context: click.Context, case_id: str):
```

Comment:
This is a CLI convention which was used in prod for a while and also is documented in AM + Atlas  as  workflow use instruction everywhere. Maybe it can be changed in another PR while also a lot of documentation is updated.